gem/migrations/0030_create_groups_and_permissions.py

```diff
@@ -0,0 +1,191 @@
+# flake8: noqa: E128
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-05-24 12:20
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.core.management.sql import emit_post_migrate_signal
+
+
+def create_groups_and_permissions(apps, schema_editor):
+    db_alias = schema_editor.connection.alias
+
+    emit_post_migrate_signal(2, False, db_alias)
+
+    Group = apps.get_model('auth.Group')
+    Permission = apps.get_model('auth.Permission')
+    GroupPagePermission = apps.get_model('wagtailcore.GroupPagePermission')
+    BannerIndexPage = apps.get_model('core.BannerIndexPage')
+    SectionIndexPage = apps.get_model('core.SectionIndexPage')
+    FooterIndexPage = apps.get_model('core.FooterIndexPage')
+    PollsIndexPage = apps.get_model('polls.PollsIndexPage')
+    SurveysIndexPage = apps.get_model('surveys.SurveysIndexPage')
+    YourWordsCompetitionIndexPage = apps.get_model(
+        'yourwords.YourWordsCompetitionIndexPage')
+
+    # remove the existing groups
+    Group.objects.all().delete()
+
+    # **** Get IndexPages ****
+    sections = SectionIndexPage.objects.all()
+    banners = BannerIndexPage.objects.all()
+    footers = FooterIndexPage.objects.all()
+    polls = PollsIndexPage.objects.all()
+    surveys = SurveysIndexPage.objects.all()
+    yourwords = YourWordsCompetitionIndexPage.objects.all()
```

Comment:
How big do you expect some of these object lists to be? Could it become a memory problem to load them all into memory upfront?

Reply:
@jpaidoussi the largest multi site cluster has 30 sites, so for that app it would be a list of 30 objects for each of the Index Pages, I don't expect it to be a problem, what do you thing?

Reply:
I don't think that would cause a problem, but it may help to use `iterator()` anyway: https://docs.djangoproject.com/en/2.0/ref/models/querysets/#iterator